src/configuration.ts

```diff
@@ -63,7 +64,10 @@ export interface IConfigurationLoadResult {
     results?: IConfigurationFile;
 }
 
-export const CONFIG_FILENAME = "tslint.json";
+// As per eslint convention, yaml config is used if present
+// over json config
+export const JSON_CONFIG_FILENAME = "tslint.json";
+export const CONFIG_FILENAMES = ["tslint.yaml", "tslint.yml", JSON_CONFIG_FILENAME];
```

Comment:
consider moving the json file to the first position to avoid unnecessary file lookups

I don't think we need to adopt the behavior of ESLint